tencentcloud/service_tencentcloud_cbs.go

```diff
@@ -226,6 +227,50 @@ func (me *CbsService) DescribeSnapshotById(ctx context.Context, snapshotId strin
 	return
 }
 
+func (me *CbsService) DescribeSnapshotByIds(ctx context.Context, snapshotIdsParam []*string) (snapshots []*cbs.Snapshot, errRet error) {
+	logId := getLogId(ctx)
+	request := cbs.NewDescribeSnapshotsRequest()
+	if len(snapshotIdsParam) == 0 {
+		return
+	}
+	request.SnapshotIds = snapshotIdsParam
+
+	offset := 0
+	pageSize := 100
+	for {
+		request.Offset = helper.IntUint64(offset)
+		request.Limit = helper.IntUint64(pageSize)
+
+		var err error
+		var response *cbs.DescribeSnapshotsResponse
+		err = resource.Retry(readRetryTimeout, func() *resource.RetryError {
+			ratelimit.Check(request.GetAction())
+			response, err = me.client.UseCbsClient().DescribeSnapshots(request)
+			if err != nil {
+				return retryError(err, InternalError)
+			}
+			return nil
+		})
+
+		if err != nil {
+			log.Printf("[CRITAL]%s api[%s] fail, request body [%s], reason[%s]\n",
+				logId, request.GetAction(), request.ToJsonString(), err.Error())
+			errRet = err
+			return
+		}
+		if len(response.Response.SnapshotSet) < 1 {
+			break
+		}
+
+		snapshots = append(snapshots, response.Response.SnapshotSet...)
+		if len(response.Response.SnapshotSet) < pageSize {
+			break
+		}
+		offset += pageSize
+	}
+	return
+}
```

Comment:
func (me *CbsService) DescribeSnapshotByIds(ctx context.Context, snapshotIdsParam []*string) (snapshots []*cbs.Snapshot, errRet error) {
	if len(snapshotIdsParam) == 0 {
		return
	}
	var (
		logId = getLogId(ctx)
		request  = cbs.NewDescribeSnapshotsRequest()
		err      error
		response *cbs.DescribeSnapshotsResponse
		offset, pageSize uint64 = 0, 100
	)
	request.SnapshotIds = snapshotIdsParam
	request.Offset = &offset
	request.Limit = &pageSize

	for {
		err = resource.Retry(readRetryTimeout, func() *resource.RetryError {
			ratelimit.Check(request.GetAction())
			response, err = me.client.UseCbsClient().DescribeSnapshots(request)
			if err != nil {
				return retryError(err, InternalError)
			}
			return nil
		})
		if err != nil {
			log.Printf("[CRITAL]%s api[%s] fail, request body [%s], reason[%s]\n",
				logId, request.GetAction(), request.ToJsonString(), err.Error())
			errRet = err
			return
		}
		snapshots = append(snapshots, response.Response.SnapshotSet...)
		if len(response.Response.SnapshotSet) < pageSize {
			break
		}
		offset += pageSize
	}
	return
}